docs/core/deploying/trimming/prepare-libraries-for-trimming.md

```diff
@@ -8,360 +8,234 @@ ms.date: 06/12/2023
-  - This ensures that the trimmer only analyzes the parts of the library's dependencies that are used. It tells the trimmer that any code that isn't part of a "root" can be trimmed if it's unused. Without this option, you would see warnings originating from _any_ part of a dependency that doesn't set `[AssemblyMetadata("IsTrimmable", "True")]`, including parts that are unused by your library.
+Setting `<IsTrimmable>true</IsTrimmable>` marks the assembly as "trimmable" and enables trim warnings. "trimmable" means the project:
 
-##### .csproj file
+* Is considered compatible with trimming.
+* Shouldn't generate trim related warnings when building. When used in a trimmed app, the assembly has its unused members trimmed in the final output.
 
-### [.NET 6](#tab/net6)
+The `IsTrimmable` property defaults to `true` when configuring a project as AOT-compatible with `<IsAotCompatible>true</IsAotCompatible>`. For more information, see [AOT-compatibility analyzers](../native-aot/index.md#aot-compatibility-analyzers).
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
-
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <!-- Replace with net5.0, or netcoreapp3.1 if testing for those frameworks -->
-    <TargetFramework>net6.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-    <!-- Prevent warnings from unused code in dependencies -->
-    <TrimmerDefaultAction>link</TrimmerDefaultAction>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <ProjectReference Include="path/to/MyLibrary.csproj" />
-    <!-- Analyze the whole library, even if attributed with "IsTrimmable" -->
-    <TrimmerRootAssembly Include="MyLibrary" />
-  </ItemGroup>
-
-</Project>
-```
+To generate trim warnings without marking the project as trim-compatible, use `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` rather than `<IsTrimmable>true</IsTrimmable>`.
 
-### [.NET 7](#tab/net7)
+### Show all warnings with test app
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
+To show all analysis warnings for a library, the trimmer must analyze the implementation:
 
-  <PropertyGroup>
-    <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
-    <PublishTrimmed>true</PublishTrimmed>
-  </PropertyGroup>
+* Of the library.
+* All dependencies the library uses.
 
-  <ItemGroup>
-    <ProjectReference Include="path/to/MyLibrary.csproj" />
-    <!-- Analyze the whole library -->
-    <TrimmerRootAssembly Include="MyLibrary" />
-  </ItemGroup>
+When building and publishing a library:
 
-</Project>
-```
+* The implementations of the dependencies aren't available.
+* The available reference assemblies don't have enough information for the trimmer to determine if they're compatible with trimming.
 
-### [.NET 8+](#tab/net8plus)
+Because of the dependency limitations, a self-contained test app which uses the library and its dependencies must be created. The test app executable includes all the information the trimmer requires to issue warning on trim incompatibilities in:
```

Comment:
```suggestion
Because of the dependency limitations, a self-contained test app which uses the library and its dependencies must be created. The test app includes all the information the trimmer requires to issue warning on trim incompatibilities in:
```

(The executable part is misleading and technically wrong as well)